### PR TITLE
complete hw5

### PR DESCRIPTION
**Pull request by RomanPutsilouski**

### CODE REVIEW

**Comment by RomanPutsilouski** [comment]
Complete HW6

**Comment by RomanPutsilouski** [comment]
1

File: Tasks/Putsilouski_Tasks/Task5/HW5.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+vyr = [1, 2, [2, 4, [[7, 8], 4, 6]]]
+
+
+def summa(vyr, sum=0):
+    for i in range(len(vyr)):
+        if type(vyr[i]) == int:
```
Comment (MikitaTsiarentsyeu) [inline Tasks/Putsilouski_Tasks/Task5/HW5.py:6]: would be better to check if type is not list, in order to support all numeric types and not just int

File: Tasks/Putsilouski_Tasks/Task6/HW6.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import mtmf
+
+vyr = input("выражение\n").replace(" ","")
+
+# vyr = "(4-3)*(12+(2-1))"
+while vyr.count(")"):
+    for y in range(len(vyr)):
+        if vyr[y] == ")":
+            for x in range(y, -1, -1):
+                if vyr[x] == "(":
```
Comment (MikitaTsiarentsyeu) [inline Tasks/Putsilouski_Tasks/Task6/HW6.py:10]: what if there is no parenthesis in the expression? like 2*4-1

File: Tasks/Putsilouski_Tasks/Task7/HW7.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Реализовать класс, который в зависимости от некоторых параметров (например, количество комнат),
+# будет "строить" лабиринт.
+# *Визуализировать построенный лабиринт.
+from random import randint
+class labirint(object):
+    def sozdanie(kol):
+        labirint_start = {}
+        def chern_lab(labirint_1):
```
Comment (MikitaTsiarentsyeu) [inline Tasks/Putsilouski_Tasks/Task7/HW7.py:8]: why do you define this function inside the sozdanie function? what practical sense in this approach?

File: Tasks/Putsilouski_Tasks/Task7/HW7.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            def dveri(komnaty, i):
```
Comment (MikitaTsiarentsyeu) [inline Tasks/Putsilouski_Tasks/Task7/HW7.py:9]: same here - this function can exist on the global level

File: Tasks/Putsilouski_Tasks/Task7/HW7.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                    dveri.append(dver_in)
+                    postroeno_dverei = postroeno_dverei + 1
+                komnaty[naz_k] = dveri
+                return komnaty
+
+            for i in range(kol):
+                labirint_1 = dveri(labirint_1, i)
+
+            return labirint_1
+        def dveri_sviaz_proverka(komnaty):
```
Comment (MikitaTsiarentsyeu) [inline Tasks/Putsilouski_Tasks/Task7/HW7.py:30]: as well as this one

**Review by MikitaTsiarentsyeu — APPROVED** [review]
HW5 - 9.0, no deprecations from my side;
HW6 - 8.5, the solution is good, but it supports not all cases;
HW7 - 8.0, it's not OOP, but not exactly procedural ether, because of poor function definitions, which refer to a functional approach, but make no practical sense here

**Pull request merged**